tests/remote_js/test/2/fileTest.js

```diff
@@ -0,0 +1,703 @@
+				charset: charset
+			}),
+			{
+				"Content-Type": "application/x-www-form-urlencoded",
+				"If-None-Match": "*"
+			}
+		);
+		Helpers.assert400(response);
+	});
+
+	// Skipped as there may or may not be an error
+	it('testAddFileFullParams', async function () {
+		this.skip();
+		let xml = await API.createAttachmentItem("imported_file", [], false, this);
+
+		let data = await API.parseDataFromAtomEntry(xml);
+		let serverDateModified = Helpers.xpathEval(xml, '//atom:entry/atom:updated');
+		await new Promise(r => setTimeout(r, 2000));
+		let originalVersion = data.version;
+		let file = "./work/file";
+		let fileContents = getRandomUnicodeString();
+		await fs.promises.writeFile(file, fileContents);
+		let hash = md5File(file);
+		let filename = "test_" + fileContents;
+		let mtime = parseInt((await fs.promises.stat(file)).mtimeMs);
+		let size = parseInt((await fs.promises.stat(file)).size);
+		let contentType = "text/plain";
+		let charset = "utf-8";
+
+		let response = await API.userPost(
+			config.userID,
+			`items/${data.key}/file?key=${config.apiKey}`,
+			implodeParams({
+				md5: hash,
+				filename: filename,
+				filesize: size,
+				mtime: mtime,
+				contentType: contentType,
+				charset: charset,
+				params: 1
+			}),
+			{
+				"Content-Type": "application/x-www-form-urlencoded",
+				"If-None-Match": "*"
+			}
+		);
+		Helpers.assert200(response);
+		Helpers.assertContentType(response, "application/json");
+		let json = JSON.parse(response.data);
+		assert.ok(json);
+		toDelete.push(hash);
+
+		let boundary = "---------------------------" + md5(Helpers.uniqueID());
+		let prefix = "";
+		for (let key in json.params) {
+			prefix += "--" + boundary + "\r\nContent-Disposition: form-data; name=\"" + key + "\"\r\n\r\n" + json.params[key] + "\r\n";
+		}
+		prefix += "--" + boundary + "\r\nContent-Disposition: form-data; name=\"file\"\r\n\r\n";
+		let suffix = "\r\n--" + boundary + "--";
+		response = await HTTP.post(
+			json.url,
+			prefix + fileContents + suffix,
+			{
+				"Content-Type": "multipart/form-data; boundary=" + boundary
+			}
+		);
+		Helpers.assert201(response);
+		
+		response = await API.userPost(
+			config.userID,
+			`items/${data.key}/file?key=${config.apiKey}`,
+			"upload=" + json.uploadKey,
+			{
+				"Content-Type": "application/x-www-form-urlencoded",
+				"If-None-Match": "*"
+			}
+		);
+		Helpers.assert204(response);
+		response = await API.userGet(
+			config.userID,
+			`items/${data.key}?key=${config.apiKey}&content=json`
+		);
+		xml = API.getXMLFromResponse(response);
+		data = await API.parseDataFromAtomEntry(xml);
+		json = JSON.parse(data.content);
+		assert.equal(hash, json.md5);
+		assert.equal(filename, json.filename);
+		assert.equal(mtime, json.mtime);
+		assert.equal(contentType, json.contentType);
+		assert.equal(charset, json.charset);
+		const updated = Helpers.xpathEval(xml, '/atom:entry/atom:updated');
+		assert.notEqual(serverDateModified, updated);
```

Comment:
This line fails. In phpunit test, this condition "should" always fail as well.

There seems to be an error with xpath in phpunit on this [line](https://github.com/zotero/dataserver/blob/17e67f46061d28b6cf080ea48215b9477f6472e9/tests/remote/tests/API/2/FileTest.php#L292): `$serverDateModified = array_get_first($xml->xpath('/atom:entry/atom:updated'));`. It makes `serverDateModified` always null, so on this [line](https://github.com/zotero/dataserver/blob/17e67f46061d28b6cf080ea48215b9477f6472e9/tests/remote/tests/API/2/FileTest.php#LL383C1-L383C1), `$this->assertNotEquals($serverDateModified, array_get_first($xml->xpath('/atom:entry/atom:updated')));` always passes because one of the values is null.

Fixing the xpath in phpunit for `//atom:entry/atom:updated` makes `serverDateModified ` not null but then it is the same as `updated` value fetched from the xml, so the assertion fails.

So this line errors, just like it should in phpunit. I notice this condition was removed in V3. Should re remove it here as well?